elements/lisk-codec/src/codec.ts

```diff
@@ -123,20 +123,33 @@ export class Codec {
 			schema,
 			message,
 		);
-		const decodedMessageCopy = objectUtils.cloneDeep(decodedMessage);
-		decodedMessageCopy[Symbol.iterator] = iterator;
+
+		const jsonMessageAsObject = this.toJSON(schema, decodedMessage);
+		return (jsonMessageAsObject as unknown) as T;
+	}
+
+	// For performance applications use encode() instead!
+	public encodeJSON(schema: Schema, message: object): Buffer {
+		const objectFromJson = this.toObject(schema, message);
+		return this.encode(schema, objectFromJson);
+	}
+
+	// eslint-disable-next-line class-methods-use-this
+	public toJSON<T = object>(schema: Schema, message: object): T {
+		const messageCopy = objectUtils.cloneDeep(message);
+		(messageCopy as IteratableGenericObject)[Symbol.iterator] = iterator;
 
 		recursiveTypeCast(
 			'toJSON',
-			decodedMessageCopy,
+			messageCopy as IteratableGenericObject,
 			(schema as unknown) as SchemaProps,
 			[],
 		);
-		return (decodedMessageCopy as unknown) as T;
+		return (messageCopy as unknown) as T;
 	}
 
-	// For performance applications use encode() instead!
-	public encodeJSON(schema: Schema, message: object): Buffer {
+	// eslint-disable-next-line class-methods-use-this
+	public toObject<T = object>(schema: Schema, message: object): T {
```

Comment:
`toObject` and passing object as a parameter does not feel good. 

I would prefer it to rather `toJSON` and `fromJSON`, What do you think @shuse2

Reply:
that is also fine with me =)

Reply:
@nazarhussain I didn't pick the names but actually toObject makes sense as it returns an object with Buffer instances and BigInts that are not compatible with JSON.stringify() or well not completely. But I'm ok with whatever name you think it's ok.

Reply:
Other than primitives everything is object to JS. And `toObject` lost the context of JSON, So would be better to associate both of co-related functions with the JSON which their intentions are.

Reply:
Also, it follows the context of NodeJS libraries. e.g. You will find `Buffer.from(string)`, `Buffer.from(array)` etc.

Reply:
Yes. I was just following what the original author for this requested. It's changed now.